File: lesson-2023-10-08/array.py
import random

pr = 1
a = []

n = int(input("Введите значение n: "))

for i in range(n):
    a.append(random.randint(-100, 59))
    print(f"{a[i]: 6}", end="")

print()

for i in range(n):
    summa = 0
    b = a[i]
    while b > 0:
        summa += abs(b % 10)
        b = b // 10
    if summa % 5 != 0:
        pr *= a[i]

print("==")
print("Ответ pr:", pr)

File: lesson-2023-10-08/array.pas
program B3Bog;
var
  a: array [1..100] of integer;
  pr, b, n, i, summa: integer;

begin
  pr := 1;
  randomize;
  write('Введите значение n: ');
  read(n);

  for i := 1 to n do
  begin
    a[i] := random(160) - 100;
    write(a[i]:6);
  end;

  writeln();

  for i := 1 to n do
  begin
    summa := 0;
    b := a[i];
    while b > 0 do
    begin
      summa := summa + abs(b mod 10);
      b := b div 10;
    end;
    if summa mod 5 <> 0 then
      pr := pr * a[i];
  end;

  writeln('==');
  writeln('Ответ pr: ', pr);
end.
